index.rst

```diff
@@ -1,61 +1,461 @@
+    +--------------------------------+-------------------------------------+
+    | ``enum``                       | The first value in the enumeration. |
+    +--------------------------------+-------------------------------------+
 
-:tocdepth: 1
+The ``@default`` annotation is used to overwrite the default values in Table 3.
 
-.. Please do not modify tocdepth; will be fixed when a new Sphinx theme is shipped.
+.. code-block:: none
+  :emphasize-lines: 4
 
-.. sectnum::
+  struct MyTopic {
+    long id; # default value is 0
+    float x;  # default value is 0
+    @default(NaN) float y; # default value is NaN
+  };
 
-.. TODO: Delete the note below before merging new content to the master branch.
+In the example above, the ``@default`` annotation overwrites the default value of the ``MyTopic.y`` field to ``NaN``.
+If this field is omited by the CSC that publishes this topic, DDS would automatically fill in the default value ``NaN`` for the application that subscribes to it.
 
-.. note::
+By default, the default value of an enumeration corresponds to the first value in the enumeration.
+The ``@default_literal`` annotation is used to select a different value in the enumeration as the default value.
+
+.. code-block:: none
+  :emphasize-lines: 3
+
+  enum Color {
+    GREEN,
+    @default_literal RED,
+    BLUE
+  };
+
+Currently, in ``ts_xml``, it is not possible to set optional fields or explicitly set default values for fields.
+An application that subscribes to a DDS topic will get what the CSCs publish or the default values presented in Table 3.
+
+Kafka
+=====
+
+We use `Apache Avro`_ to enconde messages in Kafka.
+Avro has a primitive type ``null`` that can be used to specify optional fields in derived types like ``records``, for example:
+
+.. _`Apache Avro`: https://avro.apache.org/docs/current/spec.html
+
+.. code-block:: js
+  :emphasize-lines: 6
+
+  {
+    "type": "record",
+    "name": "foo",
+    "fields" : [
+      {"name": "bar", "type": "float"},
+      {"name": "baz", "type": ["null", "float"], "default": null} // optional field baz
+    ]
+  }
+
+However, when an Avro record is deserialized, according to `this issue`_, the JSON representation will have all the fields defined, and the optional fields will have the value ``null``.
+
+.. _this issue: https://github.com/mtth/avsc/issues/118
+
+A default value can also be specified for fields.
+In ``records`` the default value for a field can be specified as follows:
+
+.. code-block:: js
+  :emphasize-lines: 6
+
+  {
+    "type": "record",
+    "name": "foo",
+    "fields" : [
+      {"name": "bar", "type": "float"},
+      {"name": "baz", "type": "float", "default": NaN} // field baz default to NaN
+    ]
+  }
+
+From this and previous sections, SAL Kafka could, in principal, convert an IDL schema with optional and default fields into an Avro schema.
+
+A field with the value ``NaN`` in a DDS record passed along to the Avro record, and that has been tested with SAL Kafka.
```

Comment:
I'm missing the sign on this.  You are saying a field with the value `NaN` *can* be passed along to the Avro record?  I think it just needs a few more words to make that explicit